crates/kitsune2/src/lib.rs

```diff
@@ -0,0 +1,318 @@
+            &self,
+            _space_id: SpaceId,
+        ) -> BoxFut<'_, K2Result<DynSpaceHandler>> {
+            Box::pin(async {
+                let (tx, rx) = tokio::sync::mpsc::channel(1);
+                let space_handler: DynSpaceHandler =
+                    Arc::new(HolochainSpaceHandler {
+                        notify_received_sender: tx,
+                    });
+                *self.notify_received_receiver.lock().await = Some(rx);
+                Ok(space_handler)
+            })
+        }
+        fn new_listening_address(&self, this_url: Url) -> BoxFut<'static, ()> {
+            *(self.url.lock().unwrap()) = this_url;
+            Box::pin(async {})
+        }
+    }
+
+    #[derive(Debug)]
+    struct HolochainSpaceHandler {
+        notify_received_sender: Sender<Bytes>,
+    }
+    impl SpaceHandler for HolochainSpaceHandler {
+        fn recv_notify(
+            &self,
+            _to_agent: AgentId,
+            _from_agent: AgentId,
+            _space: SpaceId,
+            data: Bytes,
+        ) -> K2Result<()> {
+            self.notify_received_sender.try_send(data).unwrap();
+            Ok(())
+        }
+    }
+
+    #[tokio::test]
+    async fn create_kitsune_instance() {
+        enable_tracing();
+        let signal_server = SbdServer::new(Arc::new(sbd_server::Config {
+            bind: vec!["127.0.0.1:0".to_string()],
+            ..Default::default()
+        }))
+        .await
+        .unwrap();
+        let server_url = format!("ws://{}", signal_server.bind_addrs()[0]);
+
+        // Create 2 Kitsune instances and 1 space each.
+        let mut kitsune_builder_1 =
+            default_builder().with_default_config().unwrap();
+        kitsune_builder_1.op_store = MemOpStoreFactory::create();
+        let kitsune_builder_1 = Arc::new(kitsune_builder_1);
+        kitsune_builder_1
+            .config
+            .set_module_config(&Tx5TransportModConfig {
+                tx5_transport: Tx5TransportConfig {
+                    server_url: server_url.clone().to_string(),
+                    signal_allow_plain_text: true,
+                },
+            })
+            .unwrap();
+        kitsune_builder_1
+            .config
+            .set_module_config(&K2GossipModConfig {
+                k2_gossip: K2GossipConfig {
+                    initiate_interval_ms: 100,
+                    min_initiate_interval_ms: 100,
+                    ..Default::default()
+                },
+            })
+            .unwrap();
+
+        let kitsune_handler_1 = Arc::new(HolochainKitsuneHandler {
+            url: std::sync::Mutex::new(
+                Url::from_str("ws://placeholder.url:1").unwrap(),
+            ),
+            notify_received_receiver: tokio::sync::Mutex::new(None),
+        });
+        let kitsune_1 = kitsune_builder_1
+            .kitsune
+            .create(kitsune_builder_1.clone(), kitsune_handler_1.clone())
+            .await
+            .unwrap();
```

Comment:
Instead of using the factory directly, use the [build function on the builder](https://github.com/holochain/kitsune2/blob/2428cf512c63e2b23efdb9566f0ff5d9fe34205d/crates/api/src/builder.rs#L117), so we get the extra validation and config markers.